chain/default_syncer.go

```diff
@@ -268,31 +268,44 @@ func (syncer *DefaultSyncer) syncOne(ctx context.Context, parent, next types.Tip
 		}
 	}
 
-	heavier, err := syncer.consensus.IsHeavier(ctx, next, *headTipSet, nextParentSt, headParentSt)
+	heavier, err := syncer.consensus.IsHeavier(ctx, next, headTipSet, nextParentSt, headParentSt)
 	if err != nil {
 		return err
 	}
 
 	if heavier {
-		// Gather the entire new chain for reorg comparison.
-		// See Issue #2151 for making this scalable.
-		iterator := IterAncestors(ctx, syncer.chainStore, parent)
-		newChain, err := CollectTipSetsOfHeightAtLeast(ctx, iterator, types.NewBlockHeight(uint64(0)))
-		if err != nil {
-			return err
-		}
-		newChain = append(newChain, next)
-		if IsReorg(*headTipSet, newChain) {
-			logSyncer.Infof("reorg occurring while switching from %s to %s", headTipSet.String(), next.String())
-		}
 		if err = syncer.chainStore.SetHead(ctx, next); err != nil {
 			return err
 		}
+		// Gather the entire new chain for reorg comparison and logging.
+		syncer.logReorg(ctx, headTipSet, next)
 	}
 
 	return nil
 }
 
+func (syncer *DefaultSyncer) logReorg(ctx context.Context, curHead, newHead types.TipSet) {
+	curHeadIter := IterAncestors(ctx, syncer.chainStore, curHead)
+	newHeadIter := IterAncestors(ctx, syncer.chainStore, newHead)
+	commonAncestor, err := FindCommonAncestor(curHeadIter, newHeadIter)
+	if err != nil {
+		logSyncer.Debugf("unexpected error when running FindCommonAncestor for reorg log: %s", err.Error())
```

Comment:
Debug is too quiet for a message that is very unexpected. I suggest warning (unless this is an expected path, in which case document why).